Jenkinsfiles/Jenkinsfile.run_playbook

```diff
@@ -0,0 +1,134 @@
+def private_key = ''
+def vault_pass = 'vault-pass'
+def aws_creds = 'cbj-deploy'
+def release_version = ''
+
+pipeline {
+  agent {
+    node {
+      label ''
+      customWorkspace 'blue-button-run-playbook'
+    }
+  }
+
+  parameters {
+    string(
+      defaultValue: "",
+      description: 'The playbook to run in instance. Required.',
+      name: 'PLAYBOOK'
+    )
+    string(
+      defaultValue: "",
+      description: 'The Gold Image AMI id that will be used as the base for app servers.',
+      name: 'AMI_ID'
+    )
+    string(
+      defaultValue: "",
+      description: 'The subnet ID where the build server will be launched.',
+      name: 'SUBNET_ID'
+    )
+    string(
+      defaultValue: "m3.medium",
+      description: 'The class/size of the ec2 instance to launch.',
+      name: 'INSTANCE_CLASS'
+    )
+    choice(
+      choices: 'dev\ntest',
+      description: 'The environment to deploy to. Required.',
+      name: 'ENV'
+    )
+  }
+
+  stages {
+    stage('Ensure PLAYBOOK, ENV, AMI_ID and SUBNET_ID') {
+      steps {
+        sh """
+        if [ -z "${params.PLAYBOOK}" ] || [ -z "${params.ENV}" ] || [ -z "${params.AMI_ID}" ] || [ -z "${params.SUBNET_ID}" ]
+        then
+          exit 1
+        fi
+        """
+      }
+    }
+
+    stage('Stop run on IMPL and PROD') {
+      steps {
+        sh """
+        if [ "${params.ENV}" == "impl" ] || [ "${params.ENV}" == "prod" ]
+        then
+          exit 1
+        fi
+        """
+      }
+    }
+
+    stage('Set private key file') {
+      steps {
+        script {
+          private_key = "${params.ENV}-key"
```

Comment:
It would be great to use an ephemeral SSH key for this process. I would settle for a separate key (or set of keys) meant for use with this pipeline specifically.

Reply:
Hello. Was just working to add a separate SSH key for the ephemeral SSH instance. Was wondering your thoughts on keys. I was thinking that the main goal was that the CBJ sub-project "run_playbook", Jenkins.run_playbook and playbook/run_playbook would be static/unchanged going forward. When a new playbook job is added, it would just require adding a new playbook and CBJ project (like "run_playbook (loadtest)" that uses that sub-project unchanged.

So was thinking of adding a required parameter called PLAYBOOK_SSH_KEY that would go in the parent project. I was denoting the other playbook related parameters as PB_*,  so that they could be passed in to ansible without having to put them in the Jenkins file paramters{} section. Hoping this was an OK way to do that? I think it works well and presents them for modification at "build with parameters" starting point.